### ISSUE TITLE
Add a FuzzyEq trait

### ISSUE THREAD

**Issue by cmyr** [linebender/kurbo#50]
We should have a floating-point aware equality comparison. There are crates out there that provide this logic, and I think it's [tricky enough](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/) that it's worth a dependency if somebody has done it well.

I suspect the best way to do this would be via some sort of `FuzzyEq` trait. the [float-cmp](https://github.com/mikedilger/float-cmp) crate looks like a likely candidate, but I would ideally like something that could, say, infer a good epsilon value for you based on the input values, although maybe this is a bit optimistic.

**Comment by derekdreery** [comment]
I agree about `float-cmp`. I have a fork where I started implementing `FuzzyEq` for the `kurbo` types.

**Comment by cmyr** [comment]
Yea, that sounds about right. I'd be curious to see what the API actually looks like, if you have a branch?

**Comment by derekdreery** [comment]
We talked at the xilem office hours. We decided that the best way forward is to experiment with using `float-cmp` in tests, but not expose it in the public API. We can then get a feel for how well it works before we commit to it.

**Comment by ctrlcctrlv** [comment]
Why not expose it with an option like IOF does?

**Comment by derekdreery** [comment]
Doing it in tests doesn't preclude adding it in future.
